Panic when trying to push wikipedia db

On the client:

```
$ dolt push
- Uploading...[mysql] 2025/05/07 09:30:45 packets.go:46 unexpected EOF
invalid connection
```

Server says:

```
panic: send on closed channel

goroutine 155633055 [running]:
github.com/dolthub/dolt/go/store/datas/pull.(*PullChunkTracker).addUnchecked(...)
        /Users/timsehn/dolthub/git/dolt/go/store/datas/pull/pull_chunk_tracker.go:101
github.com/dolthub/dolt/go/store/datas/pull.(*PullChunkTracker).Seen(0x1486bf25380, {0x107899dd0, 0x149f2ad31d0}, {0x68, 0xf9, 0xef, 0xd0, 0xd9, 0xa3, 0xc6, ...})
        /Users/timsehn/dolthub/git/dolt/go/store/datas/pull/pull_chunk_tracker.go:80 +0xec
github.com/dolthub/dolt/go/store/datas/pull.(*Puller).Pull.func4.2({0x68, 0xf9, 0xef, 0xd0, 0xd9, 0xa3, 0xc6, 0xdb, 0xcb, 0x43, ...}, ...)
        /Users/timsehn/dolthub/git/dolt/go/store/datas/pull/puller.go:388 +0x38
github.com/dolthub/dolt/go/libraries/doltcore/doltdb.pullHash.WalkAddrsForNBF.func1.1({{{0x0, 0x0}, 0x1090094f0, {0x14786ddb800, 0x18, 0x800}, {0x149dfebb650, 0x4, 0x4}}})
        /Users/timsehn/dolthub/git/dolt/go/store/types/ref.go:250 +0x130
github.com/dolthub/dolt/go/store/types.(*SerialMessage).walkRefs.SerialMessage.walkRefs.func1({0x68, 0xf9, 0xef, 0xd0, 0xd9, 0xa3, 0xc6, 0xdb, 0xcb, 0x43, ...})
        /Users/timsehn/dolthub/git/dolt/go/store/types/serial_message.go:584 +0x104
github.com/dolthub/dolt/go/store/types.SerialMessage.WalkAddrs.func1({0x1045a03ac?, 0x14006b0d768?}, {0x68, 0xf9, 0xef, 0xd0, 0xd9, 0xa3, 0xc6, 0xdb, ...})
        /Users/timsehn/dolthub/git/dolt/go/store/types/serial_message.go:777 +0x34
github.com/dolthub/dolt/go/store/prolly/message.walkProllyMapAddresses({0x107899e78, 0x10a865e00}, {0x14781107900?, 0x104569548?, 0x14006b0d838?}, 0x14006b0dc08)
        /Users/timsehn/dolthub/git/dolt/go/store/prolly/message/prolly_map.go:144 +0x134
github.com/dolthub/dolt/go/store/prolly/message.WalkAddresses({0x107899e78?, 0x10a865e00?}, {0x14781107900?, 0x104558bd8?, 0x30b061e38?}, 0x10a862ff0?)
        /Users/timsehn/dolthub/git/dolt/go/store/prolly/message/message.go:80 +0xf4
github.com/dolthub/dolt/go/store/types.SerialMessage.WalkAddrs({0x14781107900?, 0x1045b3c98?, 0x14006b0dc68?}, 0x1090094f0, 0x14ae25583a8)
        /Users/timsehn/dolthub/git/dolt/go/store/types/serial_message.go:776 +0x904
github.com/dolthub/dolt/go/store/types.SerialMessage.walkRefs(...)
        /Users/timsehn/dolthub/git/dolt/go/store/types/serial_message.go:579
github.com/dolthub/dolt/go/store/types.(*refWalker).walkSerialMessage(0x1478114f9e0?, 0x1090094f0, 0x14ae2558378)
        /Users/timsehn/dolthub/git/dolt/go/store/types/walk_refs.go:174 +0x100
github.com/dolthub/dolt/go/store/types.(*refWalker).walkValue(0x18?, 0x10758c920?, 0x10b5b1d01?)
        /Users/timsehn/dolthub/git/dolt/go/store/types/walk_refs.go:198 +0x194
github.com/dolthub/dolt/go/store/types.walkRefs({0x14781107900, 0x1144, 0x1144}, 0x1090094f0, 0x14ae2558378)
        /Users/timsehn/dolthub/git/dolt/go/store/types/walk_refs.go:33 +0x8c
github.com/dolthub/dolt/go/libraries/doltcore/doltdb.pullHash.WalkAddrsForNBF.func1({{0x50, 0xf2, 0xf6, 0xff, 0x7f, 0xf5, 0xe, 0xdd, 0x4b, 0xb6, ...}, ...}, ...)
        /Users/timsehn/dolthub/git/dolt/go/store/types/ref.go:245 +0xc0
github.com/dolthub/dolt/go/store/datas/pull.(*Puller).Pull.func4()
        /Users/timsehn/dolthub/git/dolt/go/store/datas/pull/puller.go:387 +0x29c
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /Users/timsehn/go/pkg/mod/golang.org/x/sync@v0.12.0/errgroup/errgroup.go:78 +0x54
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 155632339
        /Users/timsehn/go/pkg/mod/golang.org/x/sync@v0.12.0/errgroup/errgroup.go:75 +0x94
```

--- a/go/store/datas/pull/puller.go
+++ b/go/store/datas/pull/puller.go
@@ -327,7 +327,6 @@ func (p *Puller) Pull(ctx context.Context) error {
 
 	// One thread calls ChunkFetcher.Get on each batch.
 	eg.Go(func() error {
-		defer tracker.Close()
 		for {
 			toFetch, hasMore, err := tracker.GetChunksToFetch(ctx)
 			if err != nil {
@@ -347,6 +346,12 @@ func (p *Puller) Pull(ctx context.Context) error {
 
 	// One thread reads the received chunks, walks their addresses and writes them to the table file.
 	eg.Go(func() (err error) {
+		// Closing the tracker belongs here, because this is
+		// the goroutine that writes to the tracker state with
+		// |Seen| and |TickProcessed| calls.  Once this thread
+		// finishes, there will be no further writes to the
+		// tracker state.
+		defer tracker.Close()
 		defer func() {
 			cerr := rd.Close()
 			err = errors.Join(err, cerr)

